File: app/schemas/insumo.py
"""Pydantic schemas for Insumo entity."""

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class InsumoBase(BaseModel):
    nome: str = Field(..., max_length=200)
    fabricante: str | None = Field(None, max_length=120)
    unidade_medida: str = Field(..., max_length=30)
    saldo_atual: Decimal = Field(default=Decimal("0"), ge=0)
    lote: str | None = Field(None, max_length=100)
    validade: date | None = None


class InsumoCreate(InsumoBase):
    pass


class InsumoUpdate(BaseModel):
    nome: str | None = Field(None, max_length=200)
    fabricante: str | None = Field(None, max_length=120)
    unidade_medida: str | None = Field(None, max_length=30)
    saldo_atual: Decimal | None = Field(None, ge=0)
    lote: str | None = Field(None, max_length=100)
    validade: date | None = None


class InsumoResponse(InsumoBase):
    id: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/insumo.ts
/** TypeScript interfaces for Insumo entity matching Pydantic schemas. */

export interface InsumoResponse {
  id: number;
  nome: string;
  fabricante: string | null;
  unidade_medida: string;
  saldo_atual: number;
  lote: string | null;
  validade: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface InsumoCreate {
  nome: string;
  fabricante?: string | null;
  unidade_medida: string;
  saldo_atual?: number;
  lote?: string | null;
  validade?: string | null;
}

export interface InsumoUpdate {
  nome?: string | null;
  fabricante?: string | null;
  unidade_medida?: string | null;
  saldo_atual?: number | null;
  lote?: string | null;
  validade?: string | null;
}
